src/mbgl/map/map.cpp

```diff
@@ -421,11 +421,24 @@ CameraOptions Map::cameraForLatLngs(const std::vector<LatLng>& latLngs, const Ed
     // CameraOptions origin is at the top-left corner.
     centerPixel.y = viewportHeight - centerPixel.y;
 
-    options.center = latLngForPixel(centerPixel);
+    options.center = transform.screenCoordinateToLatLng(centerPixel);
     options.zoom = zoom;
     return options;
 }
 
+CameraOptions Map::cameraForLatLngs(const std::vector<LatLng>& latLngs, const EdgeInsets& padding) const {
+    return mbgl::cameraForLatLngs(latLngs, impl->transform, padding);
+}
+
+CameraOptions Map::cameraForLatLngs(const std::vector<LatLng>& latLngs, double heading, const EdgeInsets& padding) const {
+    double angle = -heading * util::DEG2RAD;
+    Transform transform(impl->transform.getState());
+    transform.setAngle(angle);
+    CameraOptions options = mbgl::cameraForLatLngs(latLngs, transform, padding);
+    options.angle = angle;
```

Comment:
The return value inherits the current pitch, but the developer may want to animate to an arbitrary pitch for the same reason that they'd want to animate to an arbitrary angle.

Reply:
Opened this as a separate issue #10115. It depends on us first getting a fix for #2259.